### PR TITLE
Tweak IOP modules printf

### PR DESCRIPTION
**Pull request by israpps**
- enforce modname as a prefix on printf's of some modules
- encase some printf inside macros, so they're not added when building non debug (left all those "Welcome" printf intact
- unify printf macros into single header for certain modules

### CODE REVIEW

**Comment by israpps** [comment]
dont forget to squash instead of common merge xd

File: iop/cdvd/cdfs/src/cdfs_iop.h [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #define _CDFS_IOP_H
 
 #ifdef DEBUG
-#define DPRINTF(args...)	printf(args)
+#define DPRINTF(args...)	printf("CDFS: "args)
```
Comment (AKuHAK) [inline iop/cdvd/cdfs/src/cdfs_iop.h:5]: why not use?
```
#define MODNAME "CDFS"
IRX_ID(MODNAME, 1, 1);

#define M_PRINTF(format, args...) \
    printf(MODNAME ": " format, ##args)
```
like here:
https://github.com/ps2dev/ps2sdk/blob/master/iop/dev9/atad/src/ps2atad.c#L40

Comment (israpps) [reply iop/cdvd/cdfs/src/cdfs_iop.h:5]: I thought that the string passed to IRX_ID() macro had to remain EXACTLY the same...

If I'm wrong, and that can be done, please let me know

Comment (israpps) [reply iop/cdvd/cdfs/src/cdfs_iop.h:5]: so...
you want me to use modname on all of them?

i would like to keep it this way

Comment (fjtrujy) [reply iop/cdvd/cdfs/src/cdfs_iop.h:5]: I think that using @AKuHAK approach will be a better solution

**Review by asmblur — APPROVED** [review]
Looks fine to me.

**Review by asmblur — APPROVED** [review]

File: iop/network/smap/src/smap.c [hunk 2]
```diff
@@ -154,12 +154,12 @@ static int InitPHY(struct SmapDriverData *SmapDrivPrivData)
 
     LinkSpeed100M = 0;
     if (EnableVerboseOutput != 0)
-        DEBUG_PRINTF("smap: Resetting PHY\n");
+        DEBUG_PRINTF("SMAP: Resetting PHY\n");
```
[the remark this answers is not in the document]
Comment (israpps) [reply iop/network/smap/src/smap.c:157]: > "SMAP: " should be removed here (many times in this file), becouse DEBUG_PRINTF also adds "SMAP: "

seems I missed that one.

Will deal with it now

**Comment by fjtrujy** [comment]
@israpps , do you have anything else pending to do in this PR?

I think you still miss something else, which is remove the `printf` function from the import list if not needed. If the IOP driver is always just using `printf` when `DEBUG`, you should remove that import as well.

I think you should rebase the PR and ensure that all the drivers follow this pattern.

Additionally, I think that is good to merge this, as it could save some KBs on the IRX generation as skipping the usage of `printf` functions (I'm not really sure as `printf` is a external function from `stdio`).

**Comment by israpps** [comment]
> @israpps , do you have anything else pending to do in this PR?
> 
> I think you still miss something else, which is remove the `printf` function from the import list if not needed. If the IOP driver is always just using `printf` when `DEBUG`, you should remove that import as well.
> 
> I think you should rebase the PR and ensure that all the drivers follow this pattern.
> 
> Additionally, I think that is good to merge this, as it could save some KBs on the IRX generation as skipping the usage of `printf` functions (I'm not really sure as `printf` is a external function from `stdio`).

Printf comes from system module `STDIO.IRX`

Some modules have a greeting printf on their start function indicating versión and other stuff...

IIRC I had nothing left to do with this PR...

I'll check it tonight if i can

**Comment by israpps** [comment]
Made a quick test

Importing from an IRX module increases binary in 16 bytes, and every imported func is another 16 byte increase

Wich means that the removal of printf implies reducing binary of at least 16 or 32 bytes (depending if IRX uses other exports from stdio).
This ofc does not consider size increase of the strings passed to printf

**Comment by fjtrujy** [comment]
> Made a quick test
> 
> Importing from an IRX module increases binary in 16 bytes, and every imported func is another 16 byte increase
> 
> Wich means that the removal of printf implies reducing binary of at least 16 or 32 bytes (depending if IRX uses other exports from stdio). This ofc does not consider size increase of the strings passed to printf

So then, is it ready to be merged? Have you rebased and checked if new drivers recently added miss this new way of print info?

@uyjulian what do you think?

Cheers.

**Comment by uyjulian** [comment]
These set of changes right now should be fine.

**Review by fjtrujy — APPROVED** [review]

**Pull request merged**